src/lang/builtins_source.ml

```diff
@@ -96,3 +96,43 @@ let () =
          Clock.force_init (fun x -> List.exists (fun y -> Oo.id x = Oo.id y) l)
        in
          Lang.list ~t:s_t (List.map (fun x -> Lang.source (x:>Source.source)) l))
+
+let () =
+  let kind = Lang.univ_t 1 in
+  add_builtin "source.dump"  ~cat:Liq
+    ~descr:"Immediately encode the whole contents of a source into a file."
+    ~flags:[Lang.Experimental]
+    [
+      "", Lang.format_t kind, None, Some "Encoding format." ;
+      "", Lang.string_t, None, Some "Name of the file.";
+      "", Lang.source_t (Lang.univ_t 1), None, Some "Source to encode";
+    ]
+    Lang.unit_t
+    (fun p ->
+       let format_val = Lang.assoc "" 1 p in
+       let format = Lang.to_format format_val in
+       let encoder =
+         try
+           Encoder.get_factory format
+         with
+         | Not_found -> raise (Lang_errors.Invalid_value (format_val, "Unsupported format"))
+       in
+       let kind = Encoder.kind_of_format format in
+       let file = Lang.to_string (Lang.assoc "" 2 p) in
+       let s = Lang.to_source (Lang.assoc "" 3 p) in
+       let encoder = encoder "source.dump" Meta_format.empty_metadata in
+       let frame = Frame.create kind in
+       let oc = open_out file in
+       Strings.iter (output_substring oc) encoder.Encoder.header;
+       s#get_ready [];
+       while s#is_ready do
+         Frame.clear frame;
+         let start = Frame.position frame in
+         s#get frame;
+         let stop = Frame.position frame in
+         let data = encoder.Encoder.encode frame start (stop-start) in
+         Strings.iter (output_substring oc) data
+       done;
+       Strings.iter (output_substring oc) (encoder.Encoder.stop ());
```

Comment:
You need to cleanup the source as well. I'd suggest the use of `output.file`, which has more advanced features (in particular in the coming `ffmpeg-video` branch, will be able to use encoder's file output capabilities to encode formats with a header such as `mp4`). 

You should build the whole `output.file(%format,"file",source)` in the code then follow the expected invokations:
- `file_output#get_ready [(file_output :> Source.source)]`
- `file_output#output_get_ready`
- `file_output#output` (many times)
- `file_output#leave (file_output :> Source.source)`

You can look at `clock.ml` for details about those calls.

I'm still unclear as to wether we will need to handle the clock situation in this case.